src/java.base/share/classes/java/lang/constant/DirectMethodHandleDesc.java

```diff
@@ -134,16 +134,10 @@ public static Kind valueOf(int refKind) {
          */
         public static Kind valueOf(int refKind, boolean isInterface) {
             int i = tableIndex(refKind, isInterface);
-            if (i >= 0 && i < TABLE.length) {
-                Kind kind = TABLE[i];
-                if (kind == null) {
-                    throw new IllegalArgumentException(String.format("refKind=%d", refKind));
-                }
-                if (kind.refKind == refKind && kind.isInterface == isInterface) {
-                    return kind;
-                }
+            if (i >= 2 && i < TABLE.length) {
+                return TABLE[i];
```

Comment:
This fix looks good.   The resulting `TABLE` has two entries per `refKind` for `isInterface` = `true` and `false`.